include/rpm/rpmts.h

```diff
@@ -359,7 +360,7 @@ rpmRC rpmtxnImportPubkey(rpmtxn txn, const unsigned char * pkt, size_t pktlen);
  * 			RPMRC_NOKEY on invalid keyid
  * 			RPMRC_FAIL on other failure
  */
-rpmRC rpmtxnDeletePubkey(rpmtxn txn, const char *keyid);
+rpmRC rpmtxnDeletePubkey(rpmtxn txn, rpmPubkey key);
```

Comment:
I'm okay with adding an alternative API for this that takes a pubkey, and I'm okay with using this name for it even, but don't *remove* the version that takes a fingerprint. I explained why in the PR that added it.

Reply:
Actually, scratch that.

Going forward, it is a requirement for rpmkeys (and the library part) to be able to delete keys without being able to construct a pubkey out of it. I've seen enough crypto related failures in the last few years that this is just non-negotiable. But, with what we have now, we have to load it into the keyring first anyhow. So okay for changing the argument to rpmPubkey, we'll add another interface around the lower-level keystore later.

Reply:
OK, this here is the minimal approach. It keeps the old function as rpmtxnDeletePubkeyByID. We can smush this into a different shape once we now what to do in the backend.

Reply:
Ok, but see https://github.com/rpm-software-management/rpm/issues/3375#issuecomment-2411652680 - I drop my case on the fingerprint/keyid API - been insisting on it for what is a wrong reason.